documentation/user-consent.md

```diff
@@ -48,122 +43,269 @@ The document details aspects regarding the consent management, which includes fo
+  else If Consent is needed and is NOT granted - Out Of Band Consent Capture (Push/SMS/other)
+    note over ExpO,User: User Interaction <br> out-of-band capture consent mechanism chosen by the Operator
+    ExpO->>BE: HTTP 200 OK {"auth_req_id": "{OperatorAuthReqId}"
+  end
+  loop Invoker polls until consent is granted or until expires. If granted in advance, token returned in first poll
+    BE->>+ExpO: POST /token <br>Credentials}<br>auth_req_id={OperatorAuthReqId}    
+    ExpO->>-BE: HTTP 200 OK <br>{"access_token": "{OperatorAccessToken}"} 
+  end  
+end
+BE->>ExpO: Access Operator CAMARA API <br> Authorization: Bearer {OperatorAccessToken}    
+ExpO->>ExpO: Decrypt OperatorAccessToken,<br>grants Access,<br>progresses request to API Backend,<br>gets API response  
+ExpO->>BE: CAMARA API Response
+Note over BE,FE: Response
+```
 
->[TO BE EDITED/COMPLETED] - Flow to obtain an access token to consume network API...
+<br>
 
-#### Client Credentials flow
+**Flow description**:
 
->[TO BE EDITED/COMPLETED] - Flow to obtain an access token to consume network API...
+First, the API invoker (which could potentially be the application BE, an aggregator, etc.) requests a 3-legged access token to the operator OGW platform. The process follows the OpenID Connect [Client-Initiated Backchannel Authentication (CIBA)](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html) flow.
 
-#### ... (other grant types)
+The API invoker has to provide in the authorization request (/bc_authorize) a login_hint with a valid user identifier together with the credentials and indicate the purpose for accessing the data (Step 1): 
 
-...
+- One option for the identifier is the public IP and (optionally – when applicable) port of the application. Other options could be the MSISDN or other identifiers. NOTE: In IoT scenarios or, in general, in those cases where the consumption device is different than the authorization device, the IP and port is the one of the consumption device for which the network capabilities will be requested/applied.
+- The login_hint is a hint regarding the user for whom authentication is being requested. It has the format `<type>:<identifier>`, for instance, "ipport:" for IP addresses or "tel:+" for phone numbers. A login_hint example for MSISDN "+346xxxyyyzzz" looks like:
 
-## Consent API
+    ```
+    POST /bc-authorize HTTP/1.1
+    Authorization: Basic {Credentials}
+    Content-Type: application/x-www-form-urlencoded
 
->[TO BE EDITED/COMPLETED] - Consent API: API to manage user consent, if required. For example when consent is captured by a third party and stored in operator. A third-party can register the consent captured from a user for a specific purpose.
+    login_hint=tel%3A%2B346xxyyyzzz&
+    … 
+    ```
+- Purpose under which the personal data associated to API consumption will be processed.
 
-## Annex: Flow source code
+The operator OGW platform will:
```

Comment:
```suggestion
The operator API Gateway Platform will:
```

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/54/commits/c7f9715fa8c597e4c9f94e8f0bbf3aecc501ead5.